WalletWasabi.Fluent/ViewModels/AddWallet/Create/ConfirmRecoveryWordsViewModel.cs

```diff
@@ -9,63 +8,155 @@
 	{
-		_confirmationWordsSourceList = new SourceList<RecoveryWordViewModel>();
-#if RELEASE
-		_isSkipEnable = Services.WalletManager.Network != Network.Main || System.Diagnostics.Debugger.IsAttached;
-#else
-		_isSkipEnable = true;
-#endif
-		var nextCommandCanExecute =
-			_confirmationWordsSourceList
+		_availableWords = new List<RecoveryWordViewModel>();
+		_words = words;
+		_mnemonic = mnemonic;
+		_walletName = walletName;
+	}
+
+	public ObservableCollectionExtended<RecoveryWordViewModel> ConfirmationWords { get; } = new();
+
+	[System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Uses DisposeWith()")]
+	protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disposables)
+	{
+		base.OnNavigatedTo(isInHistory, disposables);
+
+		ConfirmationWords.Clear();
+
+		var confirmationWordsSourceList = new SourceList<RecoveryWordViewModel>();
+
+		confirmationWordsSourceList
+			.DisposeWith(disposables)
 			.Connect()
 			.ObserveOn(RxApp.MainThreadScheduler)
-			.WhenValueChanged(x => x.IsConfirmed)
-			.Select(_ => _confirmationWordsSourceList.Items.All(x => x.IsConfirmed));
+			.Sort(SortExpressionComparer<RecoveryWordViewModel>.Ascending(x => x.Index))
+			.Bind(ConfirmationWords)
+			.OnItemAdded(x => x.Reset())
+			.Subscribe()
+			.DisposeWith(disposables);
 
 		EnableBack = true;
 
-		NextCommand = ReactiveCommand.CreateFromTask(() => OnNextAsync(mnemonic, walletName), nextCommandCanExecute);
+		CancelCommand = ReactiveCommand.Create(OnCancel);
 
-		if (_isSkipEnable)
+		var nextCommandCanExecute =
+			confirmationWordsSourceList
+			.Connect()
+			.WhenValueChanged(x => x.IsConfirmed)
+			.Select(_ => confirmationWordsSourceList.Items.All(x => x.IsConfirmed));
+
+		NextCommand = ReactiveCommand.CreateFromTask(OnNextAsync, nextCommandCanExecute);
+
+		SetSkip();
+
+		confirmationWordsSourceList.AddRange(_words);
+
+		AvailableWords =
+			confirmationWordsSourceList.Items
+									   .Select(x => new RecoveryWordViewModel(x.Index, x.Word))
+									   .OrderBy(_ => Random.Shared.Next())
+									   .ToList();
+
+		var availableWordsSourceList = new SourceList<RecoveryWordViewModel>();
+
+		availableWordsSourceList
+			.DisposeWith(disposables)
+			.Connect()
+			.WhenPropertyChanged(x => x.IsSelected)
+			.Subscribe(x => OnWordSelected(x.Sender));
+
+		availableWordsSourceList.AddRange(AvailableWords);
+
+		SetNextWord();
+
+		var enableCancel = Services.WalletManager.HasWallet();
+		SetupCancel(enableCancel: false, enableCancelOnEscape: enableCancel, enableCancelOnPressed: false);
+	}
+
+	private void SetNextWord()
+	{
+		CurrentWord =
+			CurrentWord?.Index switch
+			{
+				null => ConfirmationWords.First(),
+				int i when i >= 1 && i < ConfirmationWords.Count => ConfirmationWords[i],
+				_ => null
+			};
+
+		if (CurrentWord is { })
```

Comment:
As far as I can see having `CurrentWord` as nullable makes the code ugly.

So we could initialize it in the constructor if the list was already sorted. In order to do that. I would remove the `.Sort()` call  from `confirmationWordsSourceList`, and would do this in the constructor:

```c#
_words = words.OrderBy(x => x.Index).ToList();
_currentWord = _words.First();
```

So the null checks can be removed everywhere. And `SetNextWord()` can be simplified too:
```c#
private void SetNextWord()
{
	if (ConfirmationWords.FirstOrDefault(x => !x.IsConfirmed) is { } nextWord)
	{
		CurrentWord = nextWord;
	}

	EnableAvailableWords(true);
}
```